test/unittest/tests/block/test_block_file.cpp

```diff
@@ -0,0 +1,274 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_file]: block_open.c
+ * The block manager extent list consists of both extent blocks and size blocks. This unit test
+ * suite tests aims to test all of the allocation and frees of the extent and size block functions.
+ *
+ * The block session manages an internal caching mechanism for both block and size blocks that are
+ * created or discarded.
+ */
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include "../wrappers/mock_session.h"
+#include "../wrappers/config_parser.h"
+#include <iostream>
+
+const std::string ALLOCATION_SIZE = "512";
+const std::string BLOCK_ALLOCATION = "best";
+const std::string OS_CACHE_MAX = "0";
+const std::string OS_CACHE_DIRTY_MAX = "0";
+const std::string ACCESS_PATTERN = "random";
+const std::string DEFAULT_FILE_NAME = "test.txt";
+
+void
+validate_block_fh(WT_BLOCK *block, std::string const& name)
+{
+    REQUIRE(block->fh != nullptr);
+    REQUIRE(std::string(block->fh->name) == name);
+    REQUIRE(block->fh->file_type == WT_FS_OPEN_FILE_TYPE_DATA);
+    REQUIRE(block->fh->ref == 1);
+}
+
+void
+validate_block_config(WT_BLOCK *block, config_parser const& cp)
+{
+    std::map<std::string, std::string> const& config_map = cp.get_config_map();
+    auto it = config_map.find("allocation_size");
+    uint32_t expected_alloc_size =
+      it != config_map.end() ? std::stoi(it->second) : std::stoi(ALLOCATION_SIZE);
+    REQUIRE(block->allocsize == expected_alloc_size);
+
+    it = config_map.find("block_allocation");
+    uint32_t expected_block_allocation =
+      it != config_map.end() && it->second.compare(BLOCK_ALLOCATION) == 0 ? 0 : 1;
+
+    REQUIRE(block->allocfirst == expected_block_allocation);
+    REQUIRE(block->os_cache_max == std::stoi(config_map.at("os_cache_max")));
+    REQUIRE(block->os_cache_dirty_max == std::stoi(config_map.at("os_cache_dirty_max")));
+}
+
+void
+validate_block(std::shared_ptr<MockSession> session, WT_BLOCK *block, config_parser const& cp,
+  uint expected_ref, std::string const& name, bool readonly = false)
+{
+
+    REQUIRE(block != nullptr);
+
+    // Test Block immediate members.
+    REQUIRE(std::string(block->name) == name);
+    REQUIRE(block->objectid == WT_TIERED_OBJECTID_NONE);
+    // REQUIRE(block->compact_session_id = WT_SESSION_ID_INVALID);
+    REQUIRE(block->ref == expected_ref);
```

Comment:
This would be a good time to use CHECK so we see whether everything is wrong instead of just one field.